app/src/main/java/org/wikipedia/bridge/JavaScriptActionHandler.kt

```diff
@@ -23,4 +27,43 @@ object JavaScriptActionHandler {
     fun setMulti(context: Context, theme: String, dimImages: Boolean, collapseTables: Boolean): String {
         return context.getString(R.string.page_mh_set_multi_script, theme, dimImages, collapseTables)
     }
+
+    @JvmStatic
+    fun setFooter(context: Context, model: PageViewModel): String {
+        if (model.page == null) {
+            return "";
+        }
+
+        val showLanguagesLink = model.page!!.pageProperties.languageCount > 0
+        val showEditHistoryLink = !(model.page!!.isMainPage || model.page!!.isFilePage)
+        val lastEditText = (context.getString(R.string.last_updated_text, formatDateRelative(model.page!!.pageProperties.lastModified)))
+        val showTalkLink = !(model.page!!.title.namespace() === Namespace.TALK)
+        val showMapLink = model.page!!.pageProperties.geo != null
+
+        // TODO: page-library also supports showing disambiguation ("similar pages") links and
+        // "page issues". We should be mindful that they exist, even if we don't want them for now.
+
+        return "pagelib.c1.Footer.add(document," +
+                "    '${model.page?.displayTitle}'," +
+                "    [" +
+                (if (showLanguagesLink) "pagelib.c1.Footer.MenuItemType.languages, " else "") +
+                (if (showEditHistoryLink) "pagelib.c1.Footer.MenuItemType.lastEdited, " else "") +
+                (if (showTalkLink) "pagelib.c1.Footer.MenuItemType.talkPage, " else "") +
+                (if (showMapLink) "pagelib.c1.Footer.MenuItemType.coordinate " else "") +
+                "],   {" +
+                "        'readMoreHeading': '" + context.getString(R.string.read_more_section) + "'," +
+                "        'menuDisambiguationTitle': '" + context.getString(R.string.page_similar_titles) + "'," +
+                "        'menuLanguagesTitle': '" + context.getString(R.string.language_count_link_text, model.page?.pageProperties?.languageCount) + "'," +
```

Comment:
If the current article is available in Chinese, it will need to be updated since we automatically add either Simplified or Traditional Chinese in the "Other languages" list.